packages/react-components/src/button/src/useToggleButton.ts

```diff
@@ -1,7 +1,22 @@
+import { ForwardedRef, SyntheticEvent } from "react";
 import { isNil } from "lodash";
 import { useControllableState, useEventCallback } from "../../shared";
 
-export function useToggleButton({
+export interface UseToggleButtonProps<Shape> {
+    variant?: "solid" | "outline" | "ghost";
+    shape?: Shape;
+    active?: boolean;
+    checked?: boolean;
+    defaultChecked?: boolean;
+    value?: string | number;
+    onChange?(event: SyntheticEvent, isChecked: boolean): void;
+    onCheck?(event: SyntheticEvent, value: string | number): void;
+    forwardedRef?: ForwardedRef<any>;
+}
+
+export function useToggleButton<
+    Shape extends "pill" | "rounded" | "circular"
```

Comment:
How come you have to do this instead of:

```
shape?: "pill" | "rounded" | "circular";
```

Reply:
The idea here is to preserve the type of shape in the output variable. Meaning that in some cases, InnerToggleIconButton for instance, the allowed shape properties are circular or rounded, never pill. So if we typed the shape allowing  "pill" | "rounded" | "circular", the output props of the toggle button would also be  "pill" | "rounded" | "circular", even if we specified that the shape can only be rounded or circular

Reply:
TODO: TS commentaire